client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -223,7 +337,7 @@ function mediaButton( editor ) {
 
 			// To avoid an undesirable flicker after the image uploads but
 			// hasn't yet been loaded, we preload the image before rendering.
-			const imageUrl = MediaSerialization.deserialize( event.content ).media.URL;
```

Comment:
This would no longer work if `event.content` contains also the `[caption]` shortcode.